### PR TITLE
fix line display when zoomout in maps and refacto zoomToFeature function

### LINKED ISSUE #5826: When searching a line the focus zoom can be too large
### What happened?

When searching for a long line (like Paris-Lyon `830000`) the focus is done but the zoom is too large `z < 5`. It's an issue since tracks are not displayed at this level of zoom.

### What did you expect to happen?

Center the camera on the searched object but limit the z value to `5`.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to the "Map viewer"
2. Select an infra
3. Search for a big line

### What operating system, browser and environment are you using?

- **Browser**: Firefox v119
- **OS**: Nixos v23.05
- **Env**: Staging

### OSRD version (top right corner `Account` button > `Informations`)

2019297

### PR DESCRIPTION
**Pull request by RomainValls**
- Deleted 3 zoomToFeature functions to create one in the map helpers. 

closes #5826

### CODE REVIEW

File: front/src/common/Map/WarpedMap/core/helpers.ts [hunk 1]
```diff
@@ -175,3 +177,33 @@ export function simplifyFeature(feature: MapGeoJSONFeature): Feature {
...
+      [maxLng, maxLat],
+    ],
+    { padding: 40 }
+  );
+
+  const newViewport = {
+    ...mapViewport,
+    longitude,
+    latitude,
+    zoom: zoom > 5 ? zoom : 5,
```
Comment (flomonster) [inline front/src/common/Map/WarpedMap/core/helpers.ts:204]: You could use a max function like `max(zoom, 5) `

**Review by Yohh — COMMENTED** [review]
doesn't work in infra editor

**Comment by bloussou** [comment]
Any update on this @RomainValls ?

**Comment by RomainValls** [comment]
My bad, didn't realize it wasn't finished. I'll resolve conflicts and ping for another review.

**Review by Math-R — APPROVED** [review]
lgtm

**Pull request merged**